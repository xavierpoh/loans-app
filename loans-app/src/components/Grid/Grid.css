.repayment-detail {
  background-color: #fff;
  -webkit-box-shadow: 0 0 5px 0 rgba(141, 152, 179, 0.3);
  box-shadow: 0 0 5px 0 rgba(141, 152, 179, 0.45);
  border-radius: 4px;
  padding: 24px;
  width: 250px;
  max-width: 100%;
}

.with-margin {
  margin-right: 20px;
}

.header {
  color: #a2abb4;
  font-weight: 400;
}

.header.weekly {
  color: #3f4347;
}

.main {
  font-size: 24px;
  font-weight: 700;
  margin-top: 6px;
}

@media (max-width: 600px) {
  .with-margin {
    margin-right: 0;
    margin-bottom: 20px;
  }
  .repayment-detail {
    width: 600px;
  }
}