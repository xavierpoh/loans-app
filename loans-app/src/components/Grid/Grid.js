import React from 'react';
import './Grid.css';

const Grid = (props) => {

  function addComma(input) {
    if(input === '') {
      return '0.00';
    }
    return parseFloat(input).toLocaleString(undefined, {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2
    });
  }

  return (
    <div className={`repayment-detail ${props.hasMargin ? 'with-margin' : ''}`}>
      <div className="header">{props.header}</div>
      <div className="main">{`${props.gridData.is_money ? '$' : ''}${props.gridData.is_money? addComma(props.gridData.data) : props.gridData.data}`}</div>
    </div>
  )
}

export default Grid;