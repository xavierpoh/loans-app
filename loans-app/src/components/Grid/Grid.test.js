import Grid from './Grid';
import React from 'react';
import { mount, shallow } from 'enzyme';

const header = 'Loan Amount';
const gridData = {
  data: 10000,
  is_money: true
};

describe('Grid', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Grid header={header} gridData={gridData} />);
    expect(wrapper).toMatchSnapshot();
  })
})

test('GridComponent renders correct text inside it', () => {
  const wrapper = mount(
    <Grid header={header} gridData={gridData} />
  );
  const headerDiv = wrapper.find('.header');
  expect(headerDiv.text()).toBe('Loan Amount');

  const mainDiv = wrapper.find('.main');
  expect(mainDiv.text()).toBe('$10,000.00');
})