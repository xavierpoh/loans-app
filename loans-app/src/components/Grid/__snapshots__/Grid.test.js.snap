// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid
    gridData={
        Object {
            "data": 10000,
            "is_money": true,
          }
    }
    header="Loan Amount"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          Loan Amount
</div>,
        <div
          className="main"
>
          $10,000.00
</div>,
      ],
      "className": "repayment-detail ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loan Amount",
          "className": "header",
        },
        "ref": null,
        "rendered": "Loan Amount",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "$10,000.00",
          "className": "main",
        },
        "ref": null,
        "rendered": "$10,000.00",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            Loan Amount
</div>,
          <div
            className="main"
>
            $10,000.00
</div>,
        ],
        "className": "repayment-detail ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loan Amount",
            "className": "header",
          },
          "ref": null,
          "rendered": "Loan Amount",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "$10,000.00",
            "className": "main",
          },
          "ref": null,
          "rendered": "$10,000.00",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
